tests/integration/netdev.bats

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+	requires root
+	setup_busybox
+	# create a dummy interface to move to the container
+	ip link add dummy0 type dummy
+	ip link set up dev dummy0
+}
+
+function teardown() {
+	ip link del dev dummy0
+	teardown_bundle
+}
+
+@test "move network device to container network namespace" {
+	update_config ' .linux.netDevices |= {"dummy0": {} }
+								| .process.args |= ["ip", "link", "show", "dev", "dummy0"]'
+
+	# create a temporary name for the test network namespace
+	tmp=$(mktemp)
+	rm -f "$tmp"
+	ns_name=$(basename "$tmp")
+	# create network namespace
+	ip netns add "$ns_name"
+	ns_path=$(ip netns add "$ns_name" 2>&1 | sed -e 's/.*"\(.*\)".*/\1/')
+
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+
+	ip netns del "$ns_name"
+}
+
+@test "move network device to container network namespace and set mtu" {
+	update_config ' .linux.netDevices |= {"dummy0": {} }
+								| .process.args |= ["ip", "link", "show", "dev", "dummy0"]'
+
+	# set a custom mtu to the interface
+	ip link set mtu 1789 dev dummy0
+
+	# create a temporary name for the test network namespace
+	tmp=$(mktemp)
+	rm -f "$tmp"
+	ns_name=$(basename "$tmp")
+	# create network namespace
+	ip netns add "$ns_name"
+	ns_path=$(ip netns add "$ns_name" 2>&1 | sed -e 's/.*"\(.*\)".*/\1/')
+
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
```

Comment:
Why not just delete all of this and let runc create the netns?

Reply:
You had to do this hack, because you were checking for the netns path. But you shouldn't need that. You can use it if it's available, but if it's not, you will have to get the fd in some other way (as I suggested in the previous comment).

Reply:
added a test for that

Reply:
the problem now is to handle the restoration of the interfaces